packages/core/src/sync-historical/index.ts

```diff
@@ -535,13 +536,32 @@ export const createHistoricalSync = async (
       }),
     ).then((traces) => traces.flat());
 
-    if (traces.length > 0) {
-      await args.syncStore.insertTraces({
-        traces,
+    if (isKilled) return;
+
+    await args.syncStore.insertTraces({
```

Comment:
Notice that you removed the `if (traces.length > 0)` check, seems fine but figured I'd flag

Reply:
Yes, it's checked in `syncStore.insertTraces()`